src/components/Integrations/Shopify/Shopify.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import { Helmet } from 'react-helmet';
+import Loading from '../../Loading/Loading';
+import { InjectAppServices } from '../../../services/pure-di';
+
+const Shopify = ({ dependencies: { shopifyClient } }) => {
+  const [shops, setShops] = useState([]);
+  const [isConnected, setIsConnected] = useState(false);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const getData = async () => {
+      const result = await shopifyClient.getShopifyData();
+      if (result.expectedError && result.expectedError.cannotConnectToAPI) {
+        setError('Error: No hemos podido conectar con la Api de Shopify, vuelve a intentar luego.');
+      } else if (!result.success) {
+        setError('Error: Error inesperado.');
+      } else if (result.value.length) {
+        setIsConnected(true);
+        setShops(result.value);
+      }
```

Comment:
Creo que falta esto:

```suggestion
      } else {
        setIsConnected(false);
        setShops(result.value);
      }
```

Reply:
That's not quite right, if I remove the condition from this last if in the case that is disconnected, it fails one test. 
```
return { success: true, value:[]};
```
If there is no correct data change through out the if, by default is disconnected.

Reply:
I don't need an extra else, because is disconnected by default. Perhaps it's confusing...

Reply:
In general, having deterministic functions that do not depend on external context (original value) is a good thing.

Take into account that this function will be executed whenever React wants. So, maybe this function could be executed twice. 

Suppose this scenario: the current user is changed, I do not know, a login in another page, or something happens with current shopify integration, before there was a shop, and now there is not a shop, before was connected, and now it is not.

You are missing this scenario here.